### PR TITLE
Add Healthcheck for route53 module

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?

Adds creating health checks to boto route53 module

### What issues does this PR fix or reference?
Closese saltstack/zh#1196

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/modules/boto_route53.py [hunk 1]
```diff
@@ -340,6 +341,107 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
...
+
+    CLI Example::
+
+        salt myminion boto_route53.create_healthcheck 192.168.0.1
+        salt myminion boto_route53.create_healthcheck 192.168.0.1 port=443 hc_type=HTTPS \
+                                                      resource_path=/ fqdn=blog.saltstack.furniture
+    '''
+    if fqdn is None and ip_addr is None:
+        log.error('One of the following must be specified: fqdn or ip_addr')
+        return None
```
Comment (ryan-lane) [inline salt/modules/boto_route53.py]: I know the rest of the module returns None/True/False, but we're trying to move to a return format that always returns a dict, similar to:

```
return {'results': my_return}
```

Or in the case of failure:

```
return {'error': error_to_return}
```

This lets us cleanly return errors back to end-users through the state modules in cases of unexpected failure without throwing stacktraces or raising exceptions across module boundaries.

As an example:

https://github.com/saltstack/salt/blob/develop/salt/modules/boto_sqs.py#L104-L123

https://github.com/saltstack/salt/blob/develop/salt/states/boto_sqs.py#L118-L128

Comment (gtmanfred) [reply salt/modules/boto_route53.py]: :+1: updated

**Comment by gtmanfred** [comment]
Go Go Jenkins!

**Comment by ryan-lane** [comment]
+1 tests and everything. nice. Will you also be adding support for the state module?

**Comment by gtmanfred** [comment]
I will not be.  I was just cleaning up the WIP that Mike started here #40089

**Comment by tkwilliams** [comment]
The code looks nice and clean.  Note that the "retry on API throttling" code has recently been abstracted out to a separate function -- `_try_func(conn, func, **args)`, which you might want to leverage instead of handling it inside your function; but YMMV :)

**Comment by gtmanfred** [comment]
I do not see `_try_func` anywhere in develop or oxygen branchs.

```
$ git grep '_try_func'
$
```

File: salt/modules/boto_route53.py [hunk 2]
```diff
@@ -340,6 +341,108 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
...
+                      port=53, hc_type='TCP', resource_path='', string_match=None, request_interval=30,
+                      failure_threshold=3, retry_on_errors=True, error_retries=5):
+    '''
+    Create a Route53 healthcheck
+
+    .. versionadded:: Oxygen
+
+    ip_addr
+
+        ip address to check.  ip_addr or fqdn is required.
```
Comment (cachedout) [inline salt/modules/boto_route53.py]: `IP address`

File: salt/modules/boto_route53.py [hunk 3]
```diff
@@ -340,6 +341,108 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
...
+
+    fqdn
+
+        domain name of the endpoint to check.  ip_addr or fqdn is required
```
Comment (cachedout) [inline salt/modules/boto_route53.py]: `Domain name`

File: salt/modules/boto_route53.py [hunk 4]
```diff
@@ -340,6 +341,108 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
...
+
+        Healthcheck type.  HTTP | HTTPS | HTTP_STR_MATCH | HTTPS_STR_MATCH | TCP
+
+    resource_path
+
+        Path to check
+
+    string_match
+
+        if hc_type is HTTP_STR_MATCH or HTTPS_STR_MATCH, the string to search for in the
```
Comment (cachedout) [inline salt/modules/boto_route53.py]: `If hc_type`

File: salt/modules/boto_route53.py [hunk 5]
```diff
@@ -340,6 +341,108 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
...
+
+    failure_threshold
+
+        The number of consecutive health checks that an endpoint must pass or fail for
+        Amazon Route 53 to change the current status of the endpoint from unhealthy to
+        healthy or vice versa.
+
+    region
+
+        region endpoint to connect to
```
Comment (cachedout) [inline salt/modules/boto_route53.py]: `Region`

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by tkwilliams** [comment]
Oh, bleh, sorry.  The changes I mentioned are in https://github.com/saltstack/salt/pull/43985, which hasn't been merged yet...  Oh well, we can def retrofit later. :)

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
@gtmanfred Looks like your new test needs a little ❤️ 

Can you take a look?

- unit.modules.test_boto_route53.BotoRoute53TestCase.test_create_healthcheck
```
Error Message

'CreateHealthCheckResponse'
Stacktrace

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/moto/core/models.py", line 71, in wrapper
    result = func(*args, **kwargs)
  File "/testing/tests/unit/modules/test_boto_route53.py", line 124, in test_create_healthcheck
    del healthcheck['CreateHealthCheckResponse']['HealthCheck']['CallerReference']
KeyError: 'CreateHealthCheckResponse'
```

**Comment by gtmanfred** [comment]
Should be fixed with the last commit.

**Pull request merged**